s3-plugin-server/src/main/scala/com/gu/teamcity/ArtifactUploader.scala

```diff
@@ -31,13 +33,21 @@ class ArtifactUploader(config: S3ConfigManager, s3: S3) extends BuildServerAdapt
                   report("Not configured for uploading")
                   Continuation.BREAK
                 }
-            } recover {
+            } recoverWith {
+              case e: ResetException => {
+                runningBuild.addBuildMessage(new BuildMessage1(DefaultMessagesInfo.SOURCE_ID,
+                  DefaultMessagesInfo.MSG_BUILD_PROBLEM, Status.WARNING, new Date,
+                  s"Retrying artifact upload after error: ${e.getMessage}"))
```

Comment:
`e.getMessage` isn't always amazingly useful feedback. I'm not familiar with the style here but would it be sensible to at least log the full error so more details are available?